### PR TITLE
Upgraded test cases coverage to desired threshold of 80 percent

### PR DESCRIPTION
**Pull request by Tusharhotbit123**
1)Added test files for components with where coverage was near zero

2)Enhanced testing with edge cases where ever missing

Here is the screenshot:
Screenshot from 2025-04-03 15-52-56

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+nodeLinker: node-modules
```
Comment (shubham43MP) [inline .yarnrc.yml:1]: what is the use of this?

File: package.json [hunk 2]
```diff
@@ -53,5 +53,6 @@
   "engines": {
     "npm": ">=8.0.0",
     "node": ">=20.0.0 <=22.14.0"
-  }
+  },
+  "packageManager": "yarn@4.8.1+sha512.bc946f2a022d7a1a38adfc15b36a66a3807a67629789496c3714dd1703d2e6c6b1c69ff9ec3b43141ac7a1dd853b7685638eb0074300386a59c18df351ef8ff6"
```
Comment (shubham43MP) [inline package.json:57]: explain the changes

**Review by shubham43MP — APPROVED** [review]
